### PR TITLE
[WIP] Events

### PR DESCRIPTION
**Pull request by mt-inside**
A module for events (including an abstraction over consul's API - which returns a list of all events - to get particular ones in the sequence, e.g. the more recent). Plus some tidying up.

### CODE REVIEW

**Comment by johnhamelink** [comment]
Hello again @mt-inside :smile: Thanks for yet another great PR. Can you add some specs for the new event class you've added, as well as some documentation for the Readme?

**Comment by mt-inside** [comment]
Hi :) Yeah no problem, realised I'd missed the README just after I'd raised the PR. Specs are in progress, will take a while longer, understand you not wanting to merge without them though.

**Comment by johnhamelink** [comment]
Thanks @mt-inside. Once you submit them I'll make a new minor release with these changes!

**Comment by mt-inside** [comment]
Now with specs and updated README

File: README.md [hunk 1]
```diff
@@ -118,6 +118,39 @@ Release a lock:
...
+List all events with a certain name received by the local agent:
+
+```ruby
+Diplomat::Event.get_all('do_something')
+```
+
+Get the latest event with a certain name received by the local agent:
+
+```ruby
+Dipomat::Event.get('do_something')
```
Comment (johnhamelink) [inline README.md]: Can you fix this spelling mistake? :smile:

**Comment by johnhamelink** [comment]
[Released as version 0.8.0](https://rubygems.org/gems/diplomat/versions/0.8.0)

**Comment by mt-inside** [comment]
Argh sorry about the spelling mistake, still can't see it. Thanks for fixing.

**Comment by johnhamelink** [comment]
No worries, thanks for contributing! :trophy:

**Pull request merged**